packages/playwright-core/src/server/frames.ts

```diff
@@ -841,8 +841,12 @@ export class Frame extends SdkObject {
     return this.selectors.queryAll(selector);
   }
 
-  async queryCount(selector: string): Promise<number> {
-    return await this.selectors.queryCount(selector);
+  async queryCount(selector: string, options: any): Promise<number> {
+    try {
+      return await this.selectors.queryCount(selector, options);
+    } catch {
+      return 0;
```

Comment:
You should check `isNonRetriableError()` here. See how `isVisibleInternal()` does it. Let's add a test that `count()` throws on page closure.